common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -802,22 +827,26 @@ public static Inet6Address getByName(CharSequence ip, boolean ipv4Mapped) {
                 begin = -1;
                 break;
             case '.':
-                ++ipv4Seperators;
-                if (i - begin > IPV4_MAX_CHAR_BETWEEN_SEPARATOR
-                        || ipv4Seperators > IPV4_SEPARATORS
-                        || (ipv6Seperators > 0 && (currentIndex + compressLength < 12))
+                ++ipv4Separators;
+                tmp = i - begin; // tmp is the length of the current segment.
+                if (tmp > IPV4_MAX_CHAR_BETWEEN_SEPARATOR
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (19) used in the expression (maximum allowed 5). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1067)